### ISSUE TITLE
User actions in lobby are executed by index and can apply to the wrong user

### ISSUE THREAD

**Issue by vgaming** [wesnoth/wesnoth#5840]
### Game and System Information
 - What version of the game are you running? = 1.14.16
 - Where did you download it from? = ArchLinux
 - What OS are you running? = ArchLinux

### Describe the bug
When I try to execute an action by selecting a user in lobby, it can be applied to the wrong user if the original target disconnects from the server

### To Reproduce
1. Open Multiplayer lobby (log in)
2. Select a target user by double-clicking on somebody. A pop-up should appear
3. Without closing the popup, wait till the target user disconnects (if the target user is your second nickname, then just disconnect with the second nickname from the server)
4. With the popup window still open, execute some action like Add Friend or Send Private Message (tested on `Send Private Message`)

### Expected behavior
The action should be applied to the user I originally selected

What actually happens: the action is applied to the alphabetically next user. Note that this also applies to moderation commands, like "kick" or "ban". It is hard to undo these commands as well.

### Screenshots
![2021 06 07_09:20:19](https://user-images.githubusercontent.com/33452702/120975517-9af65980-c771-11eb-88d8-417fa4b711ff.png)

**Comment by soliton-** [comment]
Can you make:
https://github.com/wesnoth/wesnoth/blob/957496560b4b9eb8fe7d5e41db334a9f3435e7ea/src/gui/dialogs/multiplayer/lobby.cpp#L1173-L1179

into:
``` c++
 delay_playerlist_update_ = true;
 lobby_delay_gamelist_update_guard g(*this); 

 lobby_player_info dlg(*chatbox_, *info, lobby_info_); 
 dlg.show(); 
```
and see if that fixes it?

**Comment by vgaming** [comment]
@soliton- I've built the latest `master` to have a baseline for comparison, but on `master` the bug is not present.

Any opinions on what to do about the issue?

**Comment by soliton-** [comment]
Are you sure it's not present? How did you test?

**Comment by vgaming** [comment]
@soliton- I've started two `master` branch clients. I've connected both of them to the online server. With one of them, I've "selected" (popup window) the other. Then I disconnected the other. Then in the popup screen of the first client, I've made actions like "send message". A chat was opened with the "other" nickname even though that nickname has already disconnected.

Note that there was no alphabetically "next" person on the server, maybe it matters. I'll try swapping the roles of the nicknames. Maybe it also matters that the only other player is myself. I could try starting up a third client to test..
